### PR TITLE
validation api on form create and duplicate

### PR DESCRIPTION
**Pull request by abilpraju-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-3641
{36ED0025-2159-46ED-BC55-9EA5245E0571}
{B66FC599-DB50-459A-8875-197852A3D02D}

# Screenshots (if applicable)

# Notes

# Checklist
- [ ] Updated changelog
- [ ] Added meaningful title for pull request

### CODE REVIEW

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 1]
```diff
@@ -299,19 +340,172 @@ const Edit = React.memo(() => {
   const handlePublish = () => {
     console.log("publish");
   };
+  const ActionType = {
+    DUPLICATE: "DUPLICATE",
+    SAVE_AS_TEMPLATE: "SAVE_AS_TEMPLATE",
+    IMPORT: "IMPORT",
+    EXPORT: "EXPORT",
+    DELETE: "DELETE",
+  };
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:349]: CAN U MOVe this to actionModal

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 2]
```diff
@@ -299,19 +340,172 @@ const Edit = React.memo(() => {
...
+      // case ActionType.EXPORT:
+      //   setImportFormModal(true);
+      //   break;
+      // case ActionType.DELETE:
+      //     setImportFormModal(true);
+      //     break;
+      default:
+        break;
+    }
+  };
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:428]: can u move this to actionModal

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 3]
```diff
@@ -492,39 +716,94 @@ const Edit = React.memo(() => {
...
-
+      <FormBuilderModal
+        modalHeader="Duplicate"
+        nameLabel="New Form Name"
+        descriptionLabel="New Form Description"
+        showBuildForm={showBuildForm}
+        formSubmitted={formSubmitted}
+        onClose={onCloseBuildModal}
+        onAction={handleAction}
+        handleChange={handleChange}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/Item/Edit.js:764]: Can we handle this inside the reusable component itself?

File: forms-flow-web/src/components/Form/List.js [hunk 4]
```diff
@@ -381,15 +403,19 @@ const List = React.memo((props) => {
...
-                  <BuildFormModal
+                  <FormBuilderModal
+                    modalHeader="Build New Form"
+                    nameLabel="Form Name"
+                    descriptionLabel="Form Description"
                     showBuildForm={showBuildForm}
                     formSubmitted={formSubmitted}
                     onClose={onCloseBuildModal}
                     onAction={handleAction}
                     handleChange={handleChange}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Form/List.js:414]: please check whether we can handleChange from the reusable component

File: forms-flow-web/src/components/Modals/ActionModal.js [hunk 5]
```diff
@@ -3,7 +3,11 @@ import Modal from "react-bootstrap/Modal";
 import { CustomButton } from "@formsflow/components";
 import { DuplicateIcon, ImportIcon, PencilIcon, SaveTemplateIcon, CloseIcon } from "@formsflow/components";
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/ActionModal.js:5]: Add enum for ActionType here and use it as params for the methods

**Review by arun-s-aot — APPROVED** [review]

**Review by Josephalexantony-aot — APPROVED** [review]

**Pull request merged**